packages/compat/src/options.ts

```diff
@@ -92,6 +92,7 @@ const defaults = Object.assign(coreWithDefaults(), {
   extraPublicTrees: [],
   workspaceDir: null,
   optionalComponents: [],
+  packageRules: [],
```

Comment:
There was a real bug here that the new type error was revealing.

Reply:
Ah, good catch. I didn't see that, and was misinterpreting TS complaining that `options` would allow to pass something like `{ packageRules: undefined, ... }`. In fact this is possible now, and while it satisfies all type constraints, it would lead to a wrong result (returning a partial `Options`), as `Object.assign()` would ignore missing properties, but would still override the default ones with `undefined` in my example above. AFAIK TS is not able to distinguish between an object with a missing property vs. an object with a property value of `undefined`...